pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -108,46 +108,75 @@ func getConsoleTemplate(cr *v1.KieApp) v1.ConsoleTemplate {
 func getServersConfig(cr *v1.KieApp, commonConfig *v1.CommonConfig) ([]v1.ServerTemplate, error) {
 	var servers []v1.ServerTemplate
 	if len(cr.Spec.Objects.Servers) == 0 {
-		cr.Spec.Objects.Servers = []v1.KieServerSet{{Deployments: Pint(constants.DefaultKieDeployments)}}
+		cr.Spec.Objects.Servers = []v1.KieServerSet{{}}
 	}
-	names := map[string]bool{}
+	usedNames := map[string]bool{}
 	for index := range cr.Spec.Objects.Servers {
-		kieID := GenKieName(cr.Name, cr.Spec.Objects.Servers[index].Name)
-		if exists := names[kieID]; exists {
-			return []v1.ServerTemplate{}, fmt.Errorf("duplicate kieserver name %s", kieID)
+		serverSet := &cr.Spec.Objects.Servers[index]
+		if serverSet.Name == "" {
```

Comment:
serverSet.Name manipulation should happen whether the Name is empty or not. We want to prepend cr.Name to the front of any previously set serverSet.name to ensure uniqueness in a namespace.

Reply:
Same as above